src/redux/actions.ts

```diff
@@ -9,42 +9,45 @@ export const syncExperiment = async (dispatch, getState: () => AppState) => {
   // Get Latest State
   const state = getState()
   const experiment = state.experiment
-  const modules: ExperimentModule[] = Object.values(state.modules)
+  const modules: ExperimentModule[] = Object.values(state.modules).filter(
+    (mod: ExperimentModule) => mod.moduleCompleted && !mod.moduleSynced,
+  )
 
   // Finish early if experiment is offline-only
   if (experiment.offlineOnly) return
 
+  // If we have submitted all the modules thens let's flag the experiment as finished
+  if (modules.length == 0) syncExperimentEnd(experiment)
+
   // Get all modules that have been completed but not synced.
   for (const mod of modules) {
-    if (mod.moduleCompleted && !mod.moduleSynced) {
-      // Call the specific syncing method and then update the module
-      const onModuleSync = () =>
-        dispatch(
-          updateModule({
-            ...mod,
-            moduleSynced: true,
-          }),
-        )
+    // Call the specific syncing method and then update the module
+    const onModuleSync = () =>
+      dispatch(
+        updateModule({
+          ...mod,
+          moduleSynced: true,
+        }),
+      )
+
+    switch (mod.moduleType) {
+      case 'FEAR_CONDITIONING':
+        await syncFearConditioningModule(experiment, mod, onModuleSync)
+        break
 
-      switch (mod.moduleType) {
-        case 'FEAR_CONDITIONING':
-          await syncFearConditioningModule(experiment, mod, onModuleSync)
-          break
-
-        case 'BASIC_INFO':
-          await syncBasicInfoModule(experiment, mod, onModuleSync)
-          break
-
-        case 'CRITERION':
-          await syncCriterionModule(experiment, mod, onModuleSync)
-          break
-
-        default:
-          onModuleSync()
-          console.warn(
-            `Module type ${mod.moduleType} doesn't support data submission!`,
-          )
-      }
+      case 'BASIC_INFO':
+        await syncBasicInfoModule(experiment, mod, onModuleSync)
+        break
+
+      case 'CRITERION':
+        await syncCriterionModule(experiment, mod, onModuleSync)
+        break
+
+      default:
+        onModuleSync()
+        console.warn(
+          `Module type ${mod.moduleType} doesn't support data submission!`,
+        )
```

Comment:
Nitpick: This shouldn't be a warning.

Just because this is expected behaviour.